### PR TITLE
Use fetch for making API calls to MailUP

### PR DESCRIPTION
**Pull request by cloudify**
need to fix tests

### CODE REVIEW

File: src/utils/mailup.ts [hunk 1]
```diff
@@ -218,6 +227,8 @@ function toMailupAddress(
 export function MailUpTransport(
   options: IMailUpTransportOptions
 ): nodemailer.Transport {
+  const fetchAgent =
+    options.fetchAgent !== undefined ? options.fetchAgent : fetch;
```
Comment (gunzip) [inline src/utils/mailup.ts:231]: what's 'fetch' in this context, node-fetch ? (is it imported ?)

Comment (cloudify) [reply src/utils/mailup.ts:231]: you are right, I think it's just exported by `lib.dom`, not real

File: src/utils/mailup.ts [hunk 2]
```diff
@@ -109,20 +106,29 @@ function callMailUpApi(
...
+    body: JSON.stringify({ ...payload, User: creds }),
+    method
+  })
     .then(response => {
-      if (response.error) {
+      if (!response.ok) {
         return left<Error, ApiResponse>(
           new Error(
-            `Error calling MailUp API: ${response.status} - ${response.text}`
+            `Error calling MailUp API: ${response.status} - ${response.text()}`
```
Comment (gunzip) [inline src/utils/mailup.ts:120]: response.text() is a promise, this prints` [object Promise]`

File: src/utils/mailup.ts [hunk 3]
```diff
@@ -109,20 +106,29 @@ function callMailUpApi(
...
           )
         );
       }
-      return right<Error, ApiResponse>(response.body);
+      return ApiResponse.decode(response.json()).mapLeft(
```
Comment (gunzip) [inline src/utils/mailup.ts:124]: response.json is a promise

**Pull request merged**